db/migrate/20250124170335_add_source_to_collections_and_user_groups.rb

```diff
@@ -0,0 +1,9 @@
+class AddSourceToCollectionsAndUserGroups < ActiveRecord::Migration[7.0]
+  disable_ddl_transaction!
+
+  def change
+    add_reference :collections, :source, polymorphic: true, index: { algorithm: :concurrently }
+    add_reference :user_groups, :source, polymorphic: true, index: { algorithm: :concurrently }
```

Comment:
collections and user groups are pretty small tables, we probably could just build the index non-concurrently and stay safe with our ddl tx?

 I also wonder if we should make these indexes unique. There shouldn't be two collections referencing the same source entity right?

```suggestion
   safety_assured do
      add_reference :collections, :source, polymorphic: true, index: { unique: :true }
      add_reference :user_groups, :source, polymorphic: true, index: { unique: :true }
    end
```